### ISSUE TITLE
Disco Diffusion nodes not loading

### ISSUE THREAD

**Issue by NeuromindGames** [space-nuko/ComfyUI-Disco-Diffusion#6]
Hey! Having some issues getting the Custom Nodes to work with Comfyui portable. I've tried installing manually and also through the manager. The error I see when it tries to install is:

ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI-Disco-Diffusion\CLIP\clip\simple_tokenizer.py", line 6, in <module>
    import ftfy
ModuleNotFoundError: No module named 'ftfy'

There's also this in the console log:

 0.3 seconds (IMPORT FAILED): C:\path\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI-Disco-Diffusion

**Comment by WASasquatch** [comment]
See this dependency requirement addition: https://github.com/space-nuko/ComfyUI-Disco-Diffusion/pull/7

**Comment by NeuromindGames** [comment]
Still trying to get this working. I tried reinstalling both Comfyui and the Disco Diffusion custom nodes but still getting the Import Failed in the log. Here's the new error:

ModuleNotFoundError: No module named 'cv2'

**Comment by WASasquatch** [comment]
Install cv2. This are part of the requirements.txt file, are you not running it?

**Comment by NeuromindArt** [comment]
Yes I'm following the instructions as closely as I can. I'm using the ComfyUI portable for windows and I'm on Nvidia.

I just tried installing it on another computer and I get this error on that computer:

ModuleNotFoundError: No module named 'guided_diffusion'

**Comment by NeuromindGames** [comment]
I just tried reinstalling it on my main computer but still experiencing the same issue. I'm not sure how to install cv2. It's not a part of the requirements.txt file. I tried adding `opencv-python` to the requirements.txt but that didn't fix the issue. It still says: ModuleNotFoundError: No module named 'cv2'

**Comment by filliptm** [comment]
does the portable version of comfy have a virtual env to spin up? i cant find it 
^ for installing the requirements.txt

**Comment by NeuromindGames** [comment]
> does the portable version of comfy have a virtual env to spin up? i cant find it ^ for installing the requirements.txt

I was opening the CMD in the disco diffusion folder and then running: `python -m venv .env`
Once that's finished I was running: `.env\Scripts\activate`
Then one it shows .env the commandline I was putting in: `pip install -r requirements.txt`

Not sure if that's how you even do it but I found those instructions somewhere else and it seemed to work.

**Comment by filliptm** [comment]
Sweet ty, i'll give that a shot

**Comment by WASasquatch** [comment]
To install things to ComfyUI portable, you need to put in the full path to the portable EXE. .

`c:\comfyui_windows_portable\python_embeded\python.exe -s -m pip install -r "c:\comfyui_windows_portable\ComfyUI\custom_nodes\TheAwesomeCustomNode\requirements.txt"`

**Comment by NeuromindGames** [comment]
It still isn't working. I'm not sure if I'm doing it right.

I went into the disco diffusion folder in the custom nodes folder and put in `c:\comfyui_windows_portable\python_embeded\python.exe -s -m pip install -r "c:\comfyui_windows_portable\ComfyUI\custom_nodes\TheAwesomeCustomNode\requirements.txt"`

But I replaced the paths with the correct ones on my computer. It appeared to install all the requirements correctly but I still don't have the nodes in Comfyui and am getting a Failed to Import error

**Comment by WASasquatch** [comment]
> It still isn't working. I'm not sure if I'm doing it right.
> 
> I went into the disco diffusion folder in the custom nodes folder and put in `c:\comfyui_windows_portable\python_embeded\python.exe -s -m pip install -r "c:\comfyui_windows_portable\ComfyUI\custom_nodes\TheAwesomeCustomNode\requirements.txt"`
> 
> But I replaced the paths with the correct ones on my computer. It appeared to install all the requirements correctly but I still don't have the nodes in Comfyui and am getting a Failed to Import error

I think there is conflicts from installing on your system. I think to fix that you need to uninstall from both portable and system python `pip uninstall -r [requirements file]` and then reinstall only to python portable exe.

**Comment by NeuromindGames** [comment]
Can you walk me through the process or at least tell me where I'm going wrong? Here's what I'm doing:

I went into the Disco Diffusion folder and ran `pip uninstall -r [requirements file]` (I entered the correct path for my Requirements file)

Then I uninstalled everything and closed the console

Then I opened up another console inside the disco diffusion folder and typed this: 
(But switched out the paths with my actual paths)
`c:\comfyui_windows_portable\python_embeded\python.exe -s -m pip install -r "c:\comfyui_windows_portable\ComfyUI\custom_nodes\TheAwesomeCustomNode\requirements.txt"`

All it did was say that my requirements were already satisfied.

**Comment by NeuromindGames** [comment]
I did another complete uninstall and re-installed it from the Comfyui manager. It appears the nodes are working now but I keep getting out of memory errors. Even when choosing less models. I have 8gb of VRAM

**Comment by WASasquatch** [comment]
> I did another complete uninstall and re-installed it from the Comfyui manager. It appears the nodes are working now but I keep getting out of memory errors. Even when choosing less models. I have 8gb of VRAM

8GB of VRAM is not much for Disco Diffusion. It uses substantially more memory then SD. It is guiding the image process with each CLIP model select + base model

**Comment by philpax** [comment]
For reference, the full list of required dependencies can be installed with

    pip install opencv-python pandas lpips ftfy timm

for anyone else who's feeling nostalgic for Disco Diffusion. Consider also adding these to the `requirements.txt`; most of them should already be there, so I'm not sure why they failed to install through ComfyUI-Manager for me. I blame Python, as always.
